### PR TITLE
[docker-base]: Build supervisor package from source in order to update to v3.3.2

### LINKED ISSUE #762: supervisord AssertionError: Assertion failed for start.sh: RUNNING not in STARTING
On one switch during starting process, both docker bgp and docker teamd are not started successfully.

```
root@sonic:/var/log# docker logs bgp
/usr/lib/python2.7/dist-packages/supervisor/options.py:296: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
  'Supervisord is running as root and it is searching '
2017-06-27 10:23:48,055 CRIT Supervisor running as root (no user in config file)
2017-06-27 10:23:48,074 WARN Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
2017-06-27 10:23:48,375 INFO RPC interface 'supervisor' initialized
2017-06-27 10:23:48,419 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2017-06-27 10:23:48,437 INFO supervisord started with pid 1
2017-06-27 10:23:49,439 INFO spawned: 'start.sh' with pid 7
2017-06-27 10:23:50,453 INFO success: start.sh entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-06-27 10:12:34,408 INFO spawned: 'rsyslogd' with pid 20
2017-06-27 10:12:35,418 INFO success: rsyslogd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-06-27 10:12:36,208 INFO spawned: 'fpmsyncd' with pid 48
2017-06-27 10:12:37,284 INFO success: fpmsyncd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Traceback (most recent call last):
  File "/usr/bin/supervisord", line 9, in <module>
    load_entry_point('supervisor==3.0', 'console_scripts', 'supervisord')()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 360, in main
    go(options)
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 370, in go
    d.main()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 83, in main
    self.run()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 100, in run
    self.runforever()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 254, in runforever
    self.reap()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 288, in reap
    process.finish(pid, sts)
  File "/usr/lib/python2.7/dist-packages/supervisor/process.py", line 455, in finish
    self._assertInState(ProcessStates.STARTING)
  File "/usr/lib/python2.7/dist-packages/supervisor/process.py", line 179, in _assertInState
    self.config.name, current_state, allowable_states))
AssertionError: Assertion failed for start.sh: RUNNING not in STARTING

root@sonic:/var/log# docker logs teamd
/usr/lib/python2.7/dist-packages/supervisor/options.py:296: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
  'Supervisord is running as root and it is searching '
2017-06-27 10:23:48,275 CRIT Supervisor running as root (no user in config file)
2017-06-27 10:23:48,276 WARN Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
2017-06-27 10:23:48,732 INFO RPC interface 'supervisor' initialized
2017-06-27 10:23:48,735 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2017-06-27 10:23:48,740 INFO supervisord started with pid 1
2017-06-27 10:23:49,752 INFO spawned: 'start.sh' with pid 8
2017-06-27 10:23:50,789 INFO success: start.sh entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-06-27 10:12:37,278 INFO spawned: 'rsyslogd' with pid 28
2017-06-27 10:12:38,280 INFO success: rsyslogd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017-06-27 10:12:38,451 INFO spawned: 'teamd' with pid 33
2017-06-27 10:12:39,452 INFO success: teamd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Traceback (most recent call last):
  File "/usr/bin/supervisord", line 9, in <module>
    load_entry_point('supervisor==3.0', 'console_scripts', 'supervisord')()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 360, in main
    go(options)
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 370, in go
    d.main()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 83, in main
    self.run()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 100, in run
    self.runforever()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 254, in runforever
    self.reap()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 288, in reap
    process.finish(pid, sts)
  File "/usr/lib/python2.7/dist-packages/supervisor/process.py", line 455, in finish
    self._assertInState(ProcessStates.STARTING)
  File "/usr/lib/python2.7/dist-packages/supervisor/process.py", line 179, in _assertInState
    self.config.name, current_state, allowable_states))
AssertionError: Assertion failed for start.sh: RUNNING not in STARTING
```

### PR DESCRIPTION
**Pull request by jleveque**
Resolves #762.

### CODE REVIEW

**Comment by qiluo-msft** [comment]
> 

Suggest move changes on sonic-slave into individual PR, so the slave docker on the Jenkins will updated before used to check other PR.

---
Refers to: sonic-slave/Dockerfile:2 in 7be9656. [](commit_id = 7be96561fe9eda3f7e22c09324d01a7afed4d415, deletion_comment = False)

File: dockers/docker-base/Dockerfile.j2 [hunk 1]
```diff
@@ -34,25 +34,45 @@ COPY ["etc/rsyslog.conf", "/etc/rsyslog.conf"]
 COPY ["etc/rsyslog.d/*", "/etc/rsyslog.d/"]
 COPY ["root/.vimrc", "/root/.vimrc"]
 
+# Install dependencies of supervisor
```
Comment (qiluo-msft) [inline dockers/docker-base/Dockerfile.j2]: >Install dependencies of supervisor [](start = 2, length = 34)

These packages should be defined as supervisor's runtime dependencies, following other sonic component Makefile.

File: rules/docker-base.mk [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 DOCKER_BASE = docker-base.gz
 $(DOCKER_BASE)_PATH = $(DOCKERS_PATH)/docker-base
+$(DOCKER_BASE)_DEPENDS += $(SUPERVISOR)
```
Comment (qiluo-msft) [inline rules/docker-base.mk]: >SUPERVISOR [](start = 28, length = 10)

Suggest not include built packages into docker-base.
All component here are directly downloaded from debian sources server, and main updated for security update.
supervisor is likely to update more often than them, like a development package.

Comment (lguohan) [inline rules/docker-base.mk]: it is used by every docker so I think it makes sense to be included in the docker-base.

File: dockers/docker-base/Dockerfile.j2 (hunk 1 above)
Comment (lguohan) [inline dockers/docker-base/Dockerfile.j2]: use _RDEPENDS

Comment (jleveque) [inline dockers/docker-base/Dockerfile.j2]: "_RDEPENDS" is only used for installing runtime dependencies that _we build_, correct? The packages I am installing here are standard packages in the Debian repo.

**Comment by jleveque** [comment]
Regarding "Suggest move changes on sonic-slave into individual PR, so the slave docker on the Jenkins will updated before used to check other PR.":

The sonic-slave docker is now rebuilt with every PR, hence why this PR passed all 4 checks.

File: dockers/docker-base/Dockerfile.j2 (hunk 1 above)
Comment (lguohan) [inline dockers/docker-base/Dockerfile.j2]: in line 58, you have apt-get -y install -f that help to fix the dependency, so you probably do notneed line 38.

---
In reply to: [125733195](https://github.com/Azure/sonic-buildimage/pull/777#discussion_r125733195) [](ancestors = 125733195)

**Comment by lguohan** [comment]
:shipit:

**Pull request merged**